cpp/src/groupby/new_groupby.cu

```diff
@@ -190,3 +192,230 @@ gdf_error gdf_group_by(gdf_column* in_key_columns[],
+   * @Param[in]  groupby_col_indices   The column indices of the input dataset that will be grouped by
+   * @Param[out] data_cols_out         The dataset sorted by the group by columns (needs to be pre-allocated)
+   * @Param[out] group_start_indices   A column containing the starting indices of each group. Indices based off of new sort order. (needs to be pre-allocated)
+   * @Param[in]  ctxt                  The context used to control how nulls are treated in a sort and in group by
+   *   context->flag_nulls_sort_behavior
+   *    GDF_NULL_AS_LARGEST = Nulls are are treated as largest, 
+   *    GDF_NULL_AS_SMALLEST = Nulls are treated as smallest, 
+   *    GDF_NULL_AS_LARGEST_FOR_MULTISORT = Special multi-sort case any row with null is largest>
+   *   context-> flag_groupby_include_nulls 
+   *    GDF_PANDAS_STYLE = Nulls are ignored in group by keys (Pandas style),
+   *    GDF_SQL_STYLE = Nulls are treated as values in group by keys where NULL == NULL (SQL style)>
+   *
+   * @Returns gdf_error with error code on failure, otherwise GDF_SUCESS
+   */
+  /* ----------------------------------------------------------------------------*/
+gdf_error gdf_group_by_without_aggregations(gdf_size_type num_data_cols,
+                                            gdf_column** data_cols_in,
+                                            gdf_size_type num_groupby_cols,
+                                            gdf_index_type * groupby_col_indices,
+                                            gdf_column** data_cols_out,
+                                            gdf_column* group_start_indices,
+                                            gdf_context* ctxt)      
+{
+  GDF_REQUIRE((nullptr != data_cols_in), GDF_DATASET_EMPTY);
+  GDF_REQUIRE((nullptr != data_cols_in[0]), GDF_DATASET_EMPTY);
+  GDF_REQUIRE((nullptr != data_cols_out), GDF_DATASET_EMPTY);
+  GDF_REQUIRE((nullptr != data_cols_out[0]), GDF_DATASET_EMPTY);
+  GDF_REQUIRE((num_data_cols > 0), GDF_DATASET_EMPTY);
+  GDF_REQUIRE((num_groupby_cols > 0), GDF_DATASET_EMPTY);
+  GDF_REQUIRE((nullptr != groupby_col_indices), GDF_DATASET_EMPTY);
+
+
+  int32_t nrows = data_cols_in[0]->size;
+
+  // setup for order by call
+  bool group_by_keys_contain_nulls = false;
+  std::vector<gdf_column*> orderby_cols_vect(num_groupby_cols);
+  for (int i = 0; i < num_groupby_cols; i++){
+    orderby_cols_vect[i] = data_cols_in[groupby_col_indices[i]];
+    group_by_keys_contain_nulls = group_by_keys_contain_nulls || orderby_cols_vect[i]->null_count > 0;
+  }
+
+  rmm::device_vector<int32_t> sorted_indices(nrows);
+  gdf_column sorted_indices_col;
+  gdf_error status = gdf_column_view(&sorted_indices_col, (void*)(sorted_indices.data().get()), 
+                            nullptr, nrows, GDF_INT32);
+  if (status != GDF_SUCCESS)
+    return status;
+
+  if (ctxt->flag_groupby_include_nulls == GDF_SQL_STYLE || !group_by_keys_contain_nulls){  // SQL style
```

Comment:
For devs who may not be familiar with "Pandas-style" or "SQL-style", I think it would be a lot clearer to just describe what exactly you mean. E.g., instead of "Pandas-style" say something like "ignore nulls" or "discard nulls".